aten/src/ATen/NamedTensorUtils.cpp

```diff
@@ -51,6 +51,95 @@ int64_t dimname_to_position(const Tensor& tensor, Dimname dim) {
   return std::distance(names.begin(), it);
 }
 
+static void report_positional_error(
+    const Dimname& name,
+    const Dimname& other_name,
+    DimnameList names,
+    DimnameList other_names) {
+  // XXX: Can improve message by checking if names are alignable and suggesting workarounds
+  TORCH_CHECK(false,
+      "Names ", name, " and ", other_name, " do not match positionally ",
+      "from the right in names ", names, " and ", other_names, ".");
+}
+
+static void check_for_misalignment(
+    const Dimname& name,
+    DimnameList names,
+    DimnameList other_names) {
+  if (name.is_wildcard()) {
+    return;
+  }
+  auto it = std::find_if(other_names.begin(), other_names.end(),
+      [&](const Dimname& candidate) { return name.can_refer_to(candidate); });
+  // XXX: Can improve message by checking if names are alignable and suggesting workarounds
```

Comment:
why 'XXX' instead of 'TODO'?

Reply:
It's easier to type. I don't know what the difference is between "XXX" and "TODO", but I'll change it to TODO so that it actually means something.